lib/text_search/association_join.rb

```diff
@@ -25,12 +25,26 @@ def primary_key
       foreign_reflections = [ActiveRecord::Reflection::HasOneReflection,
                              ActiveRecord::Reflection::HasManyReflection]
       if foreign_reflections.include? @association.class
-        "#{@association.table_name}.#{@association.foreign_key}"
+        if @association.table_name == @model.table_name
+          "#{table_name}.#{@association.foreign_key}"
```

Comment:
Am I losing my mind? I don't see the difference between the output from the if and the else here.